apps/web/src/components/NotificationList/NotificationDownload.tsx

```diff
@@ -0,0 +1,77 @@
+import { forwardRef, MouseEventHandler, ReactNode } from "react";
+import { useIntl } from "react-intl";
+
+import { Scalars } from "@gc-digital-talent/graphql";
+import { toast } from "@gc-digital-talent/toast";
+import { errorMessages } from "@gc-digital-talent/i18n";
+
+import useAsyncFileDownload from "~/hooks/useAsyncFileDownload";
+
+import styles from "./styles";
+import { useMarkAsRead } from "./mutations";
+
+interface NotificationLinkProps {
+  id: Scalars["UUID"]["output"];
+  href: string;
+  fileName: string;
+  isUnread: boolean;
+  onRead?: () => void;
+  children: ReactNode;
+}
+
+const NotificationDownload = forwardRef<
+  HTMLAnchorElement,
+  NotificationLinkProps
+>(({ id, href, isUnread, fileName, onRead, children }, forwardedRef) => {
+  const intl = useIntl();
+  const [{ fetching }, markAsRead] = useMarkAsRead(id);
+  const [{ fetching: downloadingFile }, executeFileDownload] =
+    useAsyncFileDownload({
+      url: href,
+      fileName,
+    });
+
+  const handleClick: MouseEventHandler<HTMLAnchorElement> = async (event) => {
+    event.stopPropagation();
+    event.preventDefault();
+
+    await markAsRead().then(() => {
+      onRead?.();
+    });
+
+    await executeFileDownload().catch((err) => {
+      if (err instanceof Error && err.message === "not found") {
+        toast.error(
+          intl.formatMessage(
+            {
+              defaultMessage:
+                "{fileName} no longer exists and cannot be downloaded.",
```

Comment:
Could we switch this too "This file is older than 24 hours and is no longer available. Please request a new file. If the problem persists contact support for assistance."

Reply:
Thanks for the suggestion, updated in [update 404 file download message](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10992/commits/474081cbcb60d0caa14f851e4e238916fc64cee3)